### ISSUE TITLE
npm login does not accept email address as an user name

### ISSUE THREAD

**Issue by sathishchandran97** [npm/cli#616]
Hello all,
I have to use my email id as my user name. But npm login does not accept the email id (name@domain.com) as user name.
can anyone help me out?

Thanks in advance :)

**Comment by DanielRuf** [comment]
> I have to use my email id as my user name

This is not possible, username for npmjs.com can only contain a-z0-9 afaik.

**Comment by uxduck** [comment]
Unfortunately, this issue also prevents logins with other NPM repositories. For example, artifactory's NPM packages. I tried to login with my email on there, e.g.:
```
npm login --registry https://company.jfrog.io/artifactory/api/npm/repository-name/
```
- I entered my email, which is my Artifactory login 
Unfortunately, the error

I've had to workaround this by using `.npmrc` / basic authentication, as per https://www.jfrog.com/confluence/display/RTF/Npm+Registry. To do that, I just used `curl -u $email:$artifactory-token $registry_url`
